src/visitors/cssNamespace.js

```diff
@@ -1,21 +1,49 @@
 import * as t from 'babel-types';
 import { isStyled } from 'babel-plugin-styled-components/lib/utils/detectors';
 
+const containsInitialSelfReference = css =>
+  /^(([^\{]([\w-_,:+\s]+))*\&\s*([^\{]([\w-_,:+\s]+))*)+[\{]/g.test(css.trim());
```

Comment:
I don't mind adding tests for the particular function. I believe it will improve coverage around the behaviour adopted. Btw, shall I add the new behaviour as it is in this PR (non-breaking change)? Or should we replace the current behaviour by this bew one and release as a breaking change?